Feat todos edit

DiLiUA opened the pull request.

DiLiUA committed:
feat(todo): add todo and toggle todo is ready for test

DiLiUA committed:
feat(todo): add todo and toggle todo is ready for test

DiLiUA committed:
feat(todo): remove todo, edid todo, filtering todos are ready for test

DiLiUA committed:
Merge branch 'feat-todo-app' into feat-todos-edit

Conflicts:
	src/app/actions/todos.actions.ts
	src/app/components/todo/todo.component.html
	src/app/components/todo/todo.component.ts
	src/app/components/todos/todos.component.html
	src/app/components/todos/todos.component.ts
	src/app/effects/todos.effects.ts
	src/app/models/todo.ts
	src/app/reducers/todos.reducer.ts
	src/app/services/todos.service.ts

rychkog commented on src/app/components/todo/todo.component.ts:
```diff
@@ -0,0 +1,37 @@
+export class TodoComponent {
+  @Input() todo;
+  @Input() editableTodo;
+  @Output() editTodo = new EventEmitter();
```
type

rychkog commented on src/app/components/todo/todo.component.ts:
```diff
@@ -0,0 +1,37 @@
+  templateUrl: './todo.component.html'
+})
+export class TodoComponent {
+  @Input() todo;
```
type

rychkog commented on src/app/components/todo/todo.component.ts:
```diff
@@ -0,0 +1,37 @@
+})
+export class TodoComponent {
+  @Input() todo;
+  @Input() editableTodo;
```
type

rychkog commented on src/app/components/todo/todo.component.ts:
```diff
@@ -0,0 +1,37 @@
+    this.store.dispatch(this.actions.editTodo(todo));
+    const editableText = this.editableTodo && this.editableTodo === todo ? '' : todo.text;
+
+    this.editTodo.emit(editableText);
```
try to use it without dispatch

DiLiUA committed:
feat(todo-app): edit, remove, filtering is ready for test

Conflicts:
	package.json
	src/app/actions/todos.actions.ts
	src/app/components/todo/todo.component.html
	src/app/components/todo/todo.component.ts
	src/app/components/todos/todos.component.html
	src/app/components/todos/todos.component.ts
	src/app/effects/todos.effects.ts
	src/app/models/todo.ts
	src/app/reducers/todos.reducer.ts
	src/app/services/todos.service.ts

rychkog commented on src/app/components/todo/todo.component.html:
```diff
@@ -1,4 +1,12 @@
+    <input type="checkbox"
+           [checked]="todo.isDone"
+           (change)="toggleDoneTodo(todo)"
+           [disabled]="editableTodo ? true : false">
```
you dont need true and false - editableTodo is already a boolean

rychkog commented on src/app/components/todo/todo.component.html:
```diff
@@ -1,4 +1,12 @@
-  <input type="checkbox" [checked]="todo.isDone" (change)="toggleDoneTodo(todo)">
-  {{todo.text}}
-</label>
+<div [class.editable-todo]="editableTodo && editableTodo === todo ? true: false">
```
same as below

rychkog commented on src/app/components/todo/todo.component.ts:
```diff
@@ -1,22 +1,32 @@
 export class TodoComponent {
-  @Input() todo: Todo;
+  @Input()  todo: Todo;
+  @Input()  editableTodo;
```
type

rychkog replied:
and default value

rychkog commented on src/app/components/todos/todos.component.ts:
```diff
@@ -2,32 +2,82 @@ import { Component } from '@angular/core';
+    this.textTodo = this.editableTodo ? todo.text : '';
+  }
+
+  showAllTodos(filter: string): void {
```
those 3 methods are the same - move logic from template to here and use ngFor in the template to generate a list of filters applied

rychkog commented on src/app/components/todos/todos.component.ts:
```diff
@@ -2,32 +2,70 @@ import { Component } from '@angular/core';
+    this.textTodo = this.editableTodo ? todo.text : '';
+  }
+
+  filterTodosByFilter(filter: string) {
```
just filterTodos

rychkog commented on src/app/actions/todos.actions.ts:
```diff
@@ -10,6 +11,16 @@ export class TodosActions {
+  static EDIT_TODO = 'Edit todo';
+  static UPDATE_TODO = 'Update todo';
+  static UPDATE_TODO_SUCCESS = 'Update todo success';
+  static FILTER_TODOS_BY_FILTER = {
```
filterTodos
